src/nupic/algorithms/Connections.cpp

```diff
@@ -85,7 +85,18 @@ Synapse Connections::createSynapse(const Segment& segment,
   vector<SynapseData>& synapses = cells_[segment.cell.idx].segments[segment.idx].synapses;
   if (synapses.size() == maxSynapsesPerSegment_)
   {
-    NTA_THROW << "Cannot create synapse: segment has reached maximum number of synapses.";
+    vector<SynapseData>::iterator minPermSynIt = min_element(
```

Comment:
Instead of `minPermSynIt`, I would use a variable name like `minSynapseIterator`.